client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -58,46 +61,95 @@ public ReaderGroupConfigBuilder disableAutomaticCheckpoints() {
         * Add a stream and its associated start {@link StreamCut} and end {@link StreamCut} to be read by the
         * readers of a ReaderGroup.
         *
-        * @param scopedStreamName Scoped Name of the Stream.
-        * @param startStreamCut Start {@link StreamCut}
-        * @param endStreamCut End {@link StreamCut}
+        * @param scope Scope of the stream, if it is not specified then {@link ReaderGroup#getScope()} is used for the scope.
+        * @param streamName Stream name.
+        * @param startStreamCut Start {@link StreamCut}.
+        * @param endStreamCut End {@link StreamCut}.
         * @return Reader group config builder.
         */
-       public ReaderGroupConfigBuilder stream(final String scopedStreamName, final StreamCut startStreamCut, final StreamCut endStreamCut) {
-           final Stream stream = Stream.of(scopedStreamName);
+       public ReaderGroupConfigBuilder stream(final String scope, final String streamName,
+                                              final StreamCut startStreamCut, final StreamCut endStreamCut) {
+
+           Exceptions.checkNotNullOrEmpty(streamName, "streamName");
+
+           final String scopedStreamName;
+           if (StringUtils.isBlank(scope)) {
+               scopedStreamName = streamName;
+           } else {
+               scopedStreamName = new StringBuilder(scope).append('/').append(streamName).toString();
+           }
 
            if (startingStreamCuts == null) {
                startingStreamCuts = new HashMap<>();
            }
-           this.startingStreamCuts.put(stream, startStreamCut);
+           this.startingStreamCuts.put(scopedStreamName, startStreamCut);
 
            if (endingStreamCuts == null) {
                endingStreamCuts = new HashMap<>();
            }
-           this.endingStreamCuts.put(stream, endStreamCut);
+           this.endingStreamCuts.put(scopedStreamName, endStreamCut);
 
            return this;
        }
 
+       /**
+        * Add a stream and its associated start {@link StreamCut} and end {@link StreamCut} to be read by the
+        * readers of a ReaderGroup. {@link ReaderGroup#getScope()} is used for the scope of the stream.
+        *
+        * @param streamName Stream name.
+        * @param startStreamCut Start {@link StreamCut}.
+        * @param endStreamCut End {@link StreamCut}.
+        * @return Reader group config builder.
+        */
+       public ReaderGroupConfigBuilder stream(final String streamName, final StreamCut startStreamCut, final StreamCut endStreamCut) {
+           return stream(null, streamName, startStreamCut, endStreamCut);
```

Comment:
You could pass the default scope name rather than null here.

Reply:
Passing null because the ReaderGroupConfig does not know the default Scope for the ReaderGroup. It is passed while creating the ReaderGroupManager `io.pravega.client.admin.ReaderGroupManager#withScope(java.lang.String, java.net.URI)`

Reply:
The semantics here is confusing. If I understand it right, in the case the scope is null, we assume that the stream name is scoped as per line 76. But, if we have given a default scope to the reader group manager, then why are we not making use of it? In fact, it makes sense to have a default scope for the reader group, not the manager, unless the default scope of the manager is also propagated to the reader groups.